portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -68,12 +76,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String comment = getUserComment(request, "comment-input", "");
+    String message = getUserComment(request, "comment-input", "");
     long timestamp = System.currentTimeMillis();
 
+    Document doc =
+        Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+    double score = sentiment.getScore();
+    languageService.close();
```

Comment:
Extract that to a function (maybe `calculateScore`?) with some documentation and below you can do something like `commentEntity.setProperty(SCORE, calculateScore(...)) `